compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -548,11 +548,10 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
       def isTopLevelDefinitionInSource(sym: Symbol) =
         !sym.is(Package) && !sym.name.isPackageObjectName &&
         (sym.owner.is(Package) || (sym.owner.isPackageObject && !sym.isConstructor))
-      if !sym.hasAnnotation(defn.ExperimentalAnnot)
-        && (ctx.settings.experimental.value && isTopLevelDefinitionInSource(sym))
-        || (sym.is(Module) && sym.companionClass.hasAnnotation(defn.ExperimentalAnnot))
-      then
-        sym.addAnnotation(Annotation(defn.ExperimentalAnnot, sym.span))
+      if sym.is(Module) then
+        sym.companionClass.getAnnotation(defn.ExperimentalAnnot).foreach(sym.addAnnotation)
```

Comment:
I don't think it is safe to take an annotation tree and put it in two different places in the AST. Consider creating an accurate copy instead.

Reply:
Added some helper methods in `Annotations.ExperimentalAnnotation`.